src/psbt.h

```diff
@@ -56,6 +60,10 @@ struct PSBTInput
     CScriptWitness final_script_witness;
     std::map<CPubKey, KeyOriginInfo> hd_keypaths;
     std::map<CKeyID, SigPair> partial_sigs;
+    std::map<std::vector<unsigned char>, std::vector<unsigned char>> ripemd160_preimages;
+    std::map<std::vector<unsigned char>, std::vector<unsigned char>> sha256_preimages;
+    std::map<std::vector<unsigned char>, std::vector<unsigned char>> hash160_preimages;
+    std::map<std::vector<unsigned char>, std::vector<unsigned char>> hash256_preimages;
```

Comment:
In 9ff68e38862106c827d40ab09731bd75a59d140c "psbt: implement hash preimages fields"

Hashes are almost always represented using `uint256`/`uint160` internally, so we should key by those instead.

Reply:
I've made them byte vectors because that's how they are represented as byte arrays in the Miniscript satisfier ([1](https://github.com/sipa/miniscript/blob/2e0734da14e85fb53cd5a96b98cfb7004e289202/bitcoin/script/miniscript.h#L326-L327), [2](https://github.com/sipa/miniscript/blob/2e0734da14e85fb53cd5a96b98cfb7004e289202/bitcoin/script/miniscript.h#L677-L696)).  Keying by each hash type would thus effectively result in >=2 needless copies (once when creating them here and n times when using them with the satisfier).

Reply:
I don't think that really matters. Copying/creating/comparing a uint160/uint256 is very cheap (it's literally just 20 or 32 bytes in memory), while vector creation/copying is slow (it involves memory allocations, and some 40 bytes of overhead).